### PR TITLE
Add terraform completion

### PR DESCRIPTION
**Pull request by gkze**
Recreating https://github.com/robbyrussell/oh-my-zsh/pull/3499

CC @pmoust @robbyrussell @dlsniper

### CODE REVIEW

File: plugins/terraform/_terraform [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    'get:Download and install modules for the configuration'
+    'graph:Create a visual graph of Terraform resources'
+    'init:Initializes Terraform configuration from a module'
+    'output:Read an output from a state file'
+    'plan:Generate and show an execution plan'
+    'pull:Refreshes the local state copy from the remote server'
+    'push:Uploads the local state to the remote server'
+    'refresh:Update local state file against real resources'
+    'remote:Configures remote state management'
+    'taint:Manually forcing a destroy and recreate on the next plan/apply'
```
Comment (pmoust) [inline plugins/terraform/_terraform]: minor, I 'd keep the alphabetical order (like TF does as well)

File: plugins/terraform/_terraform [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+)
+
+__apply() {
+    _arguments \
+        '-backup=[(path) Path to backup the existing state file before modifying. Defaults to the "-state-out" path with ".backup" extension. Set to "-" to disable backup.]' \
+        '-input=[(true) Ask for input for variables if not directly set.]' \
+        '-no-color[If specified, output will not contain any color.]' \
+        '-refresh=[(true) Update state prior to checking for differences. This has no effect if a plan file is given to apply.]' \
+        '-state=[(path) Path to read and save state (unless state-out is specified). Defaults to "terraform.tfstate".]' \
+        '-state-out=[(path) Path to write state to that is different than "-state". This can be used to preserve the old state.]' \
```
Comment (pmoust) [inline plugins/terraform/_terraform]: `-target` is missing

File: plugins/terraform/_terraform [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        '-var-file=[(path) Set variables in the Terraform configuration from a file. If "terraform.tfvars" is present, it will be automatically loaded if this flag is not specified.]'
+}
+
+__get() {
+    _arguments \
+        '-update=[(false) If true, modules already downloaded will be checked for updates and updated if necessary.]'
+}
+
+__graph() {
+    _arguments \
```
Comment (pmoust) [inline plugins/terraform/_terraform]: few args missing

**Comment by pmoust** [comment]
Looks great overall, minor stuff missing. one could go through https://www.terraform.io/docs/commands/index.html to check for more.

imho, it's great to have this merged in. If an update is due (and it's bound to be since TF is only v0.6.1 and wildly adopted) the community will offer it pretty fast.

**Comment by dlsniper** [comment]
I'll send a PR against the repo @gkze has shortly fixing the above issues and doing a read on all the commands / options to be sure none is missed.

File: plugins/terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+## atom
+
+Plugin for Terraform, a tool from Hashicorp for managing infrastructure safely and efficiently.
+
+### Requirements
+
+ * [Terraform](https://teraform.io/)
```
Comment (mcornella) [inline plugins/terraform/README.md]: typo -> terraform.io

**Comment by dlsniper** [comment]
I've sent the PR with the changes but I have one doubt about it. Could someone please help me with the [push](https://www.terraform.io/docs/commands/remote.html) command as it's a bit more tricky: ```terraform remote SUBCOMMAND [options]``` where SUBCOMMAND is one of these: config, pull and push and then the options apply. Thank you.

**Comment by gkze** [comment]
Thank you again @dlsniper and @pmoust ! I just merged the v0.6.1 updates into my fork

**Comment by gkze** [comment]
@dlsniper not sure, googling right now :)

**Comment by mcornella** [comment]
Let's pull in what we have at least.
/cc @robbyrussell

**Pull request merged**